### PR TITLE
Support fieldName with dots

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

I am targeting this branch, because BC.

When working on https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1241, I thought about moving the support for fieldName with `.` in the `getFieldValue` instead. This provide support for embedded/custom getters for every persistence bundle.

## Changelog

```markdown
### Added
- Added support for fieldName with dot for `BaseFieldDescription::getFieldValue()`
```

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]

**Review by wbloszyk — DISMISSED** [review]

File: tests/Admin/BaseFieldDescriptionTest.php [hunk 1]
```diff
@@ -162,101 +161,109 @@ public function testAdmin(): void
...
+        $mock = $this->getMockBuilder(\stdClass::class)->addMethods([$method])->getMock();
 
-        $description->setOption('virtual_field', true);
-        $description->getFieldValue($mock, 'fake');
+        $mock->method($method)->willReturn(42);
+        $this->assertSame(42, $description->getFieldValue($mock, 'fake_field_value'));
+        $this->assertSame(42, $description->getFieldValue($mock, 'fakeFieldValue'));
+    }
+
+    public function getFieldValueWithFieldNameDataProvider(): iterable
```
Comment (greg0ire) [inline tests/Admin/BaseFieldDescriptionTest.php:248]: I don't remember which prefix we use in Sonata (if we still use one), but I think it's great to document the return value of data providers to match it against the signature of tests that use them.

```suggestion
    /**
     * @return iterable<array{string}>
     */
    public function getFieldValueWithFieldNameDataProvider(): iterable
```

**Review by greg0ire — COMMENTED** [review]
Do any docs need to be added, since this is a new feature?

File: tests/Admin/BaseFieldDescriptionTest.php (hunk 1 above)
Comment (VincentLanglet) [reply tests/Admin/BaseFieldDescriptionTest.php:251]: Done

**Review by wbloszyk — APPROVED** [review]

**Comment by VincentLanglet** [comment]
I don't think doc is needed here @greg0ire ; `fieldValue` will be a protected method that you can use in order to implement `getValue` ; it's more a dev-feature, than an end-user feature.

This dev-feature, will then help to create patch for persistence bundle (with the support of custom getters or embedded field) ; and the changelog will be enough.

**Pull request merged**